assets/javascripts/discourse/connectors/editor-preview/d-templates.js

```diff
@@ -1,28 +1,36 @@
+import Component from "@ember/component";
 import { action } from "@ember/object";
-import { getOwnerWithFallback } from "discourse/lib/get-owner";
+import { service } from "@ember/service";
+import { classNames } from "@ember-decorators/component";
 
 const SELECTOR_EDITOR_PREVIEW =
   "#reply-control .d-editor-preview-wrapper > .d-editor-preview";
 
-export default {
-  setupComponent(args, component) {
-    component.setProperties({
-      templatesVisible: false,
-      model: getOwnerWithFallback(this).lookup("service:composer").model,
-    });
+@classNames("d-templates")
+export default class DTemplatesEditorPreview extends Component {
+  static shouldRender(args, component) {
+    return !component.site.mobileView;
```

Comment:
```suggestion
  static shouldRender(args, context) {
    return !context.site.mobileView;
```
In the olden-days we used to pass the PluginOutlet component instance here. But nowadays, it's just a "context" object with a few services.